tendrl/gluster_integration/objects/volume/atoms/delete/__init__.py

```diff
@@ -83,13 +86,30 @@ def run(self):
             )
             return False
 
-        NS.etcd_orm.client.delete(
-            "clusters/%s/Volumes/%s" % (
-                NS.tendrl_context.integration_id,
-                self.parameters['Volume.vol_id']
-            ),
-            recursive=True
-        )
+        # Intentionally keeping sleep with high value and blocking
+        # one and not the gevent.sleep(). Make sure no sync etc
+        # happens for this time and then execute etcd delete for volume
+        time.sleep(30)
```

Comment:
put a check in NS.gdeploy_plugin.delete_volume to block till volume is actually deleted in sds, 

time.sleep doesnt belong here